pkg/teeth/github.go

```diff
@@ -81,14 +73,66 @@ func (gh *GitHub) CreateInvocationCommentWithRetry(ctx context.Context, prNum in
 	backoff := retryFunc(retryMinWaitDuration)
 	backoff = retry.WithMaxRetries(retryMaxAttempts, backoff)
 	if err := retry.Do(ctx, backoff, func(ctx context.Context) error {
+		var err error
 		resp, err := gh.commentPRFunc(ctx, gh.config.Owner, gh.config.Repo, prNum, invocationComment)
+
+		statusCode := 0
+		if resp != nil {
+			statusCode = resp.StatusCode
+		}
+
+		if resp != nil && resp.Body != nil {
+			defer resp.Body.Close()
+		}
+
+		if shouldRetry(statusCode, err) {
+			// Just in case err is nil but status code is not 2XX
+			return retry.RetryableError(fmt.Errorf("retrying error with status %d: %w", statusCode, err))
+		}
+
 		if err != nil {
-			return err
+			return fmt.Errorf("non-retryable error response: %w", err)
 		}
-		defer resp.Body.Close()
-		return convertRetryable(resp.StatusCode)
+
+		if statusCode == http.StatusNotFound || statusCode == http.StatusGone {
+			return fmt.Errorf("resource does not exist for %s/%s/pull/%d", gh.config.Owner, gh.config.Repo, prNum)
+		}
+		// Read the entire response.
+		body, err := io.ReadAll(io.LimitReader(resp.Body, 2<<20)) // 2mb
+		if err != nil {
+			return fmt.Errorf("failed to read response body: %w", err)
+		}
+		bodyStr := string(bytes.TrimSpace(body))
+
+		if statusCode != http.StatusCreated {
+			return fmt.Errorf("non-201 response: %v", bodyStr)
+		}
+		return nil
 	}); err != nil {
 		return fmt.Errorf("failed to create invocation comment: %w", err)
 	}
+
 	return nil
 }
+
+func shouldRetry(statusCode int, err error) bool {
+	// Retry if rate-limited. See https://github.com/google/go-github#rate-limiting.
+	if GitHubErrRateLimit.Is(err) || GitHubErrSecondaryRateLimit.Is(err) {
```

Comment:
I don't think this is what you want. It is testing for equality while you want to test for type.

See the implmentation: https://github.com/google/go-github/blob/eb3bed00ca77f6e718f034d07008e6f7cfd93b17/github/github.go#L1162 It is checking for RetryAfter field matching.

If you look at docs for the library, it reccomends handling using type assertions instead https://pkg.go.dev/github.com/google/go-github/v58/github#hdr-Rate_Limiting